src/libraries/Microsoft.PowerFx.Core/Public/Values/InMemoryRecordValue.cs

```diff
@@ -11,7 +11,7 @@
 namespace Microsoft.PowerFx.Types
 {
     // Represent record backed by known list of values. 
-    internal class InMemoryRecordValue : RecordValue
+    internal class InMemoryRecordValue : RecordValue, IMutationCopy
```

Comment:
>IMutationCopy [](http://example.com/codeflow?start=54&length=13)

Is there a case where  a RecordValue should not implement this?

Reply:
I am assuming that there are other kinds of records, such as marshaled objects. In those cases, the objects should not be copied and treated as a reference, or mutations should not be allowed, or it should implement IMutationCopy or sone other interface to facilitate copying when mutated. I figured we could figure this all out later, this PR bring surgical and leaving these other cases unchanged, but we can have this conversation now if you would prefer.

Reply:
I've moved this interface to RecordValue with an implementation that copies any record to an InMemoryRecord.  There are test scenarios where the table/record will override these interface implementations, as an extreme example of where a different implementation may be desired.   It is also possible that the DataverseRecordValue will have it's own implementation.  But, bottom line, no there is no record value that should not be copyable.